src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,52 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<byte> inputValue = value;
+
+            // We will ignore the leading and trailing occurrences of the white space in the json check.
+            int start = 0;
+            while (start < inputValue.Length && char.IsWhiteSpace(Convert.ToChar(inputValue[start])))
+            {
+                start++;
+            }
+            int end = inputValue.Length - 1;
+            while (end >= start && char.IsWhiteSpace(Convert.ToChar(inputValue[end])))
+            {
+                end--;
+            }
+            int len = end - start + 1;
+
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json string, Json value true, Json value false, Json number.
+            // Our logic here is that, for example, we check if the first and the last bytes are '{' and '}', respectively.
+            if (len > 0 && (
```

Comment:
The allow-list approach makes me uncomfortable.   For example, this would consider `null` invalid and thus rule out valid patterns like:

```json
"someKey" : null
```

What problem are we trying to solve here?   Why do we need this level of validation rather than just writing out the bytes?   `Utf8JsonWriter` will validate that the JSON is structurally valid by default, unless you're setting the [SkipValidation](https://docs.microsoft.com/dotnet/api/system.text.json.jsonwriteroptions.skipvalidation?view=net-6.0#system-text-json-jsonwriteroptions-skipvalidation) option.

Reply:
Good call out on `null`.

Json and String are very simple to deal with, the challenge that we are attempting to handle here are the other primitives namely `null`, `number`, `bool`.

If we want to only use BinaryData for `AnyObject` we can safely just serialize as json which is the current behavior.  If we want to expand its use to `Any` as well we must deal with these other cases also.

Deserialization is easy as no interpretation happens we simply store the raw bytes in the binary data and let the customer choose how to interpret them.  Serialization is where we now need to know how to write out what was given to us.

Reply:
Added `null` to the check list.